cupy/core/_routines_indexing.pyx

```diff
@@ -152,6 +152,22 @@ cdef ndarray _ndarray_choose(ndarray self, choices, out, mode):
     return out
 
 
+cdef ndarray _ndarray_compress(ndarray self, condition, axis, out):
+    a = self
+
+    if numpy.isscalar(condition):
+        raise ValueError('condition must be a 1-d array')
+
+    if not isinstance(condition, ndarray):
+        condition = core.array(condition, dtype=int)
+        if condition.ndim != 1:
+            raise ValueError('condition must be a 1-d array')
```

Comment:
We do not accept numpy arrays as argument in general.
I think we can just ensure `isinstance(condition, ndarray)`.

Reply:
I am not sure if I understand what you said. Doesn't `ndarray` mean only cupy ndarrays.

Reply:
Yes, that's correct. `ndarray` is `cupy.ndarray` in this context.

Reply:
If we just assert that instance is of `ndarray`, then we might not be able to take a python list of booleans as input. That's why I thought I might convert it to cupy ndarray (which also has a side-effect of converting numpy arrays also to cupy ndarrays). Otherwise, I would need to check manually for a list of booleans.

Reply:
We do not accept list of booleans as well.

Reply:
Sorry, as `take` accepts both lists and NumPy arrays as `indices`, I think `condition` can support them as well.